### PR TITLE
Fixes #1092

### LINKED ISSUE #1092: Error in IE: "Unable to get value of the property 'warn': object is null or undefined"
There is no `console` object in IE (at least 8-9) if console is not enabled.

### PR DESCRIPTION
**Pull request by Invis1ble**
This commit should fix error in IE 8-9.

### CODE REVIEW

**Comment by trentmwillis** [comment]
Thanks for this PR. Would you mind actually making this a bit more robust? I'm thinking we should have a logging singleton which abstracts away these guards. Something like:

```js
const Logger = {
  log(...args) {
    if (global.console) {
      console.log(...args);
    }
  }
};
```

Then in our code we can simply do:

```js
Logger.log('foo');
// or
Logger.warn('bar');
```

File: src/assert.js [hunk 1]
```diff
@@ -57,8 +57,10 @@ class Assert {
 	// Exports test.push() to the user API
 	// Alias of pushResult.
 	push( result, actual, expected, message, negative ) {
-		console.warn( "assert.push is deprecated and will be removed in QUnit 3.0." +
-			" Please use assert.pushResult instead (http://api.qunitjs.com/pushResult/)." );
+		if ( console ) {
```
Comment (leobalter) [inline src/assert.js]: We should add a comment in the line above `// Supports IE9, not throwing if console is not active. Ref gh-1092`

**Comment by trentmwillis** [comment]
> I like the idea from @trentmwillis to introduce a Logger api but that might come as a follow up work.

Sounds good to me. I'll do that after this lands.

File: src/assert.js (hunk 1 above)
Comment (Invis1ble) [inline src/assert.js]: I've added comment. Sorry for multiple commits.

**Comment by trentmwillis** [comment]
Thanks for the contribution @Invis1ble!

**Pull request merged**